### ISSUE TITLE
ARM/aarch64: match minimal `i8042` emulation done on x86_64

### ISSUE THREAD

**Issue by acatangiu** [firecracker-microvm/firecracker#1339]
Firecracker does not have an `i8042` device on `aarch64`. We could add the same level of emulation we do on `x86_64` that supports:
- the cpu reset event
- emulating Ctrl+Alt+Del keys combo

**Comment by pyrito** [comment]
Hi, can @mi-yu go ahead and get assigned to this issue?

**Comment by mi-yu** [comment]
To follow up, do you have more specifics on how to go about the emulation for `aarch64`? I'm looking at the code here: https://github.com/firecracker-microvm/firecracker/blob/master/devices/src/legacy/i8042.rs but I'm not sure what is `x64` specific.

**Comment by dianpopa** [comment]
Hi @mi-yu ,
The i8042 is a keyboard driver that is x86 specific (you cannot enable it for a arm64 kernel config). You can try do a `make menuconfig` inside a linux kernel on an arm64 and x86_64 machine in parallel. If you go to "Device drivers -> input device support -> hardware i/o ports" you will see that on the x86_64 you get the "i8042 PC keyboard controller" option while on the arm64 machine you do not.
What we could do for arm64 in this scenario is to implement an GPIO controller which is 			used in virtualized ARM systems to provide injection of keycodes through the linux gpio-keys driver without the need for a more complex device (i.e. a keyboard controller). 
We could use the gpio device to send a combination of CTRL+ALT+DELETE, just like we do on x86_64 by implementing the i8042.
Fortunately enough, the linux kernel offers support for the ARM PrimeCell GPIO (PL061) which is a AMBA (Advanced Microcontroller Bus Architecture) compliant device. In Firecracker we already have the RTC which is also an AMBA device implementing the PL031 specification. 
So the next step here as far as introducing ctrl-alt-del functionality on aarch64 would be to try to implement the pl061 [specification](http://infocenter.arm.com/help/topic/com.arm.doc.ddi0190b/DDI0190.pdf). Take a look at the RTC implementation (devices->src->legacy) for a little bit of inspiration.

**Comment by acatangiu** [comment]
@mi-yu @pyrito are you still interested in contributing here?

**Comment by pyrito** [comment]
@acatangiu apologies for the late response. I don't think I will have the time to contribute to this issue in the near future. I'll unassign myself. Thanks!

**Comment by sandreim** [comment]
Closing, this is a duplicate of #2046
